contractcourt/chain_arbitrator.go

```diff
@@ -341,9 +341,22 @@ func (c *ChainArbitrator) Start() error {
 				pCache:    c.cfg.PreimageDB,
 				signer:    c.cfg.Signer,
 				isOurAddr: c.cfg.IsOurAddress,
-				markChanClosed: func() error {
-					// TODO(roasbeef): also need to pass in log?
-					return c.resolveContract(chanPoint, nil)
+				notifyChanClosed: func() error {
```

Comment:
Where does the chain watcher mark the channel as fully closed? Even in that case, the `resolveContract` methods needs to be called in order to stop the channel arbitrator as well.

Reply:
It also deletes the state of the chain arbitrator itself. As is now, this state will be left lingering after the channel has been closed cooperatively.

Reply:
I still can't see where this is happening (`arbLog` will always be `nil`):

```go
func (c *ChainArbitrator) resolveContract(chanPoint wire.OutPoint,
	arbLog ArbitratorLog) error {

	log.Infof("Marking ChannelPoint(%v) fully resolved", chanPoint)

	// First, we'll we'll mark the channel as fully closed from the PoV of
	// the channel source.
	err := c.chanSource.MarkChanFullyClosed(&chanPoint)
	if err != nil {
		log.Errorf("ChainArbitrator: unable to mark ChannelPoint(%v) "+
			"fully closed: %v", chanPoint, err)
		return err
	}

	if arbLog != nil {
		// Once this has been marked as resolved, we'll wipe the log
		// that the channel arbitrator was using to store its
		// persistent state. We do this after marking the channel
		// resolved, as otherwise, the arbitrator would be re-created,
		// and think it was starting from the default state.
		if err := arbLog.WipeHistory(); err != nil {
			return err
		}
	}

	c.Lock()
	delete(c.activeChannels, chanPoint)

	chainWatcher, ok := c.activeWatchers[chanPoint]
	if ok {
		chainWatcher.Stop()
	}
	c.Unlock()

	return nil
}
```